### PR TITLE
Implement #1070

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Issue #, if available
#1070 

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

### CODE REVIEW

File: cedar-policy-validator/src/cedar_schema/err.rs [hunk 1]
```diff
@@ -372,6 +372,9 @@ pub enum ToJsonSchemaError {
     /// Invalid type name
     #[error("this uses a reserved namespace or typename: {}", .0.node)]
     ReservedName(Node<SmolStr>),
+    /// Use reserved JSON schema keywords
+    #[error("this uses a reserved JSON schema keyword: {}", .0.node)]
+    ReservedJsonSchemaKeyword(Node<SmolStr>),
```
Comment (cdisselkoen) [inline cedar-policy-validator/src/cedar_schema/err.rs:377]: I think these keywords are also reserved in the Cedar syntax?  So the error message could just say "reserved schema keyword"

File: cedar-policy-validator/src/cedar_schema/test.rs [hunk 2]
```diff
@@ -1859,6 +1862,57 @@ mod translator_tests {
...
     }
+
+    #[test]
+    fn reserved_json_schema_keyword() {
+        let schema = cedar_schema_to_json_schema(
+            parse_schema(
+                r#"
+               type Entity = Long;
+        "#,
+            )
```
Comment (cdisselkoen) [inline cedar-policy-validator/src/cedar_schema/test.rs:1873]: Can we also get tests for `namespace N { type Entity = Long; }` and similar?

File: cedar-policy-validator/src/cedar_schema/to_json_schema.rs [hunk 3]
```diff
@@ -162,6 +162,12 @@ fn convert_namespace(
     Ok((ns_name, def))
 }
 
+// Test if this id is a reserved JSON schema keyword.
+// Issue: https://github.com/cedar-policy/cedar/issues/1070
+fn is_reserved_json_schema_keyword(id: &UnreservedId) -> bool {
```
Comment (cdisselkoen) [inline cedar-policy-validator/src/cedar_schema/to_json_schema.rs:167]: Here as well, these are reserved in both syntaxes right? so we can just say "reserved schema keyword"

**Review by cdisselkoen — APPROVED** [review]
Looks good, just some naming nits.

Also I believe this fixes #1094.

File: cedar-policy-validator/src/cedar_schema/err.rs (hunk 1 above)
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/cedar_schema/err.rs:377]: Done.

File: cedar-policy-validator/src/cedar_schema/to_json_schema.rs (hunk 3 above)
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/cedar_schema/to_json_schema.rs:167]: Done.

File: cedar-policy-validator/src/cedar_schema/test.rs (hunk 2 above)
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/cedar_schema/test.rs:1873]: Done.

**Review by andrewmwells-amazon — APPROVED** [review]

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**